proto/matrix.proto

```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+
+message Vector {
+    int32 size = 1;
+    repeated double data = 2 [packed = true];
+}
+
+message Matrix {
+    int32 rows = 1;
+    int32 cols = 2;
+    repeated double data = 3 [packed = true];
+    bool rowmajor = 4;
```

Comment:
Will we ever shift from row-major to column-major? Why is this bool necessary?

Reply:
I think Eigen uses colmajor and numpy rowmajor, so we should keep track of that. Fixing it now